server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/ACVRDownloadByCounty.java

```diff
@@ -62,57 +69,98 @@ public String endpointName() {
    */
   @Override
   public String endpoint(final Request the_request, final Response the_response) {
-    final Set<String> county_set = the_request.queryParams();
     String result = "";
     int status = HttpStatus.OK_200;
     
-    try {
-      final OutputStream os = SparkHelper.getRaw(the_response).getOutputStream();
-      final BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(os, "UTF-8"));
+    if (validateParameters(the_request)) {
+      final Set<Integer> county_set = new HashSet<Integer>();
+      for (final String s : the_request.queryParams()) {
+        try {
+          county_set.add(Integer.valueOf(s));
+        } catch (final NumberFormatException e) {
+          // cannot happen because we validated the parameters
+        }
+      }
       final Set<CastVoteRecord> matches = getMatching(county_set);
       if (matches == null) {
-        result = "Unable to fetch records from database";
         status = HttpStatus.INTERNAL_SERVER_ERROR_500;
+        result = "Error retrieving records from database";
```

Comment:
Yes, that's right. Knowing what the exception was is immaterial because it, like all generic Hibernate persistence exceptions, is completely unrecoverable.